src/abi/export.rs

```diff
@@ -161,15 +170,15 @@ impl Abi {
                 exports.push(Instr::LowerObject(ret, ptr));
             }
             AbiType::Option(ty) => {
-                let var = gen.gen_num(NumType::U8);
+                let var = gen.gen_num(self.iptr());
                 let some = gen.gen((&**ty).clone());
                 rets.push(var.clone());
                 let mut some_instr = vec![];
                 self.export_return(some.clone(), gen, &mut some_instr, rets);
                 exports.push(Instr::LowerOption(ret, var, some, some_instr));
             }
             AbiType::Result(ty) => {
-                let var = gen.gen_num(NumType::U8);
+                let var = gen.gen_num(self.iptr());
```

Comment:
this won't solve the issue. it means that u8's are broken on wasm

Reply:
So, we have to use a i32 to sent over the ffi boundary with wasm. I guess we can use an u8 for dart?

Reply:
probably doesn't matter much as it's passed in a cpu register. dart won't run on a 8bit or 16bit machine anyway